### DISCUSSION TITLE
Are let bindings in F# classes supported?

### DISCUSSION THREAD

**Discussion by asik** [xunit/xunit#2365] [Question] [answered]
I noticed that this t:est is actually discovered and ran:
```fsharp
type public MyClass() =
    [<Fact>]
    let MyTest () =
        ....
```
I didn't expect it to, since it's compiled to a private method. Apparently XUnit is inquisitive enough to look at private members for test methods. Is this a behavior we can rely on or should we use public members instead?

**Comment by bartelink** [comment, marked as answer]
Yes, `private` Test Methods have always been supported by xUnit (and other FWs, more detail [from me] here: https://stackoverflow.com/a/16214704/11635)

I''d venture its pretty much idiomatic to prefer `let`-bound functions over `member`s for tests written in F#

**Reply by bradwilson** [reply]
Yep, this is definitely supported and intentional.

The usage that we wanted to enable here was for people to be able to ship private test methods in publicly released code such that a test runner could be deployed to validate production code. The ability to use `let` in F# is just a freebie that comes along for the ride. 😄
